### PR TITLE
Modernize project and make it buildable

### PR DESCRIPTION
**Pull request by rzo1**
# What does this PR do?

I checked out the repository to work on https://github.com/sweble/sweble-wikitext/issues/90 and wasn't able to build the project due to different issues.

Therefore, I went ahead to fix them:

- Move parent back to latest released version 3.0.8
- Manually overrides some outdated plugins / configurations
- Skip some checks as they cannot deal with Java 11 (due to old parent)
- Updates language level to 11
- Fix some deprecations

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
 	<parent>
 		<groupId>de.fau.cs.osr</groupId>
 		<artifactId>tooling</artifactId>
-		<version>3.0.9-SNAPSHOT</version>
+		<version>3.0.8</version>
```
Comment (wetneb) [inline pom.xml:26]: That looks curious to me. I think it is normal practice to set the version to the next patch number followed by `-SNAPSHOT` to differentiate the development branch from an official release. Why did you need to do that?

Comment (rzo1) [reply pom.xml:26]: IMHO, it is good practice to rely on a **non** SNAPSHOT **parent** for subsequent project for a few reasons:

- SNAPSHOT artifacts can change, resulting in unreproducable builds or CI failures (as the parent can change, devs can have local builds, if there is no central snapshot repo defined)
- Contributors have to build the `tooling` project in https://github.com/sweble/osr-common before they can work on this project (each time, because there is no central snapshot repo or process to deploy snapshots to oss sonatype)
- In our case, there seem to be no activity on the parent project anymore, so it seems to be unrealistic to get a `3.0.9` release artifact to Maven Central soon (happy to be corrected)

For the actual project `sweble-wikitext` , the version is still `3.1.10-SNAPSHOT` indicating a development version.

Comment (rzo1) [reply pom.xml:26]: So in general I agree, that `-SNAPSHOT` needs to be added to indicate a development version (for the actual modules inside a project). However, I think, that a parent, which resides outside of the actual project, should be a released artifact to ease development and to avoid the obstacles mentioned above.

Comment (wetneb) [reply pom.xml:26]: That makes a lot of sense to me, thanks for explaining!

**Review by wetneb — APPROVED** [review]

**Comment by wetneb** [comment]
I guess all we need to merge this is just to get the CI to pass, no?

**Comment by rzo1** [comment]
The CI failed with something unrelated afair. Maybe you can retrigger?

**Comment by wetneb** [comment]
I don't have the option to re-trigger, perhaps because the last build is too old. Could you perhaps make a tiny irrelevant change? :-D

**Comment by rzo1** [comment]
@wetneb Can you approve the workflow? ;-)

**Comment by rzo1** [comment]
Great. Don't think that 

```bash
Caused by: org.eclipse.jgit.errors.RevWalkException: Walk failure.
	at org.eclipse.jgit.revwalk.RevWalk.iterator(RevWalk.java:1259)
	at pl.project13.jgit.DescribeCommand.findCommitsUntilSomeTag(DescribeCommand.java:404)
	... 32 more
Caused by: org.eclipse.jgit.errors.MissingObjectException: Missing commit 721db2b0560630faa0ae9e7693c800fc9e8dcd54
	at org.eclipse.jgit.internal.storage.file.WindowCursor.open(WindowCursor.java:149)
	at org.eclipse.jgit.revwalk.RevWalk.getCachedBytes(RevWalk.java:883)
	at org.eclipse.jgit.revwalk.RevCommit.parseHeaders(RevCommit.java:145)
	at org.eclipse.jgit.revwalk.PendingGenerator.next(PendingGenerator.java:[148](https://github.com/sweble/sweble-wikitext/actions/runs/8642953682/job/23695614231#step:5:149))
	at org.eclipse.jgit.revwalk.StartGenerator.next(StartGenerator.java:183)
	at org.eclipse.jgit.revwalk.RevWalk.next(RevWalk.java:421)
	at org.eclipse.jgit.revwalk.RevWalk.iterator(RevWalk.java:1257)
	... 33 more
```

is actually related.

**Comment by wetneb** [comment]
I do think it is related: you have removed `fetch-depth: 0` from the git cloning options in the CI. Can you try adding it back?

**Comment by rzo1** [comment]
@wetneb thanks for the hint, seems to work now

**Comment by mawiesne** [comment]
Thx @wetneb
